app/models/concerns/profile_searching.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+# Methods that are related to profile searching
+module ProfileSearching
+  extend ActiveSupport::Concern
+
+  included do
+    scoped_search on: %i[id name ref_id account_id compliance_threshold
+                         external parent_profile_id]
+    scoped_search relation: :hosts, on: :id, rename: :system_ids
+    scoped_search relation: :hosts, on: :name, rename: :system_names
+    scoped_search on: :has_test_results, ext_method: 'test_results?',
+                  only_explicit: true, operators: ['=']
+    scoped_search on: :canonical, ext_method: 'canonical?', only_explicit: true,
+                  operators: ['=']
+
+    scope :canonical, lambda { |canonical = true|
```

Comment:
Notice the added param here: `Profile.canonical` or `Profile.canonical(true)` are equivalent. `Profile.canonical(false)` is the negative case. The same is true of `Profile.external` and `Profile.has_test_results`